src/chat/chat.gateway.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  WebSocketGateway,
+  SubscribeMessage,
+  MessageBody,
+  ConnectedSocket,
+  WebSocketServer,
+} from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
+import { ChatService } from './chat.service';
+import { Logger, UseGuards } from '@nestjs/common';
+import { createChatServerDto } from './message/dto/create-chatserver.dto';
+import { SendMessageDto } from './message/dto/send-message.dto';
+import { WsAuthGuard } from '@app/common/guards/ws-auth.guard';
+import { User } from '@app/common';
+import { AuthService } from 'src/auth/auth.service';
+
+@UseGuards(WsAuthGuard)
+@WebSocketGateway({ namespace: 'chat' })
+export class ChatGateway {
+  @WebSocketServer()
+  server: Server;
+
+  user: User | null | undefined;
+
+  constructor(
+    private readonly chatService: ChatService,
+    private readonly authService: AuthService,
+  ) {}
+
+  @SubscribeMessage('create')
+  async handleCreateChat(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() data: createChatServerDto,
+  ) {
+    const auth_token = client.handshake.headers.authorization;
+    Logger.log('Received createChat event with user: ' + data);
+    Logger.log(`auth token is: ${auth_token}`, 'ChatGateway');
+    const { userId } = await this.authService.extractUserFromToken(auth_token);
+    Logger.log('Received createChat event with customerId: ' + userId);
+    if (!userId) {
+      Logger.error('No customerId provided ' + userId);
+    }
+    const chat = await this.chatService.createChat({
+      customerId: userId,
+    });
+    Logger.log('Chat created:', chat);
+    client.emit('chatCreated', chat);
+  }
```

Comment:
Try a more dynamic way to know what users are in the chat. Perhaps add more logic to the CurrentUser decorator to fit this websocket aswell.

Reply:
I tried for days and didnt succeded, I decided that its not worth my time, if you want we can try again together but I think that its better to focus on other things, believe me when I say I tried this ALOT of time